### PR TITLE
Prevent unpublished posts to be retrieved

### PR DESCRIPTION
**Pull request by quaxsze**
Fix https://github.com/opendatateam/udata/issues/2361

### CODE REVIEW

File: udata/core/post/api.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ class PostsAPI(API):
     def get(self):
         '''List all posts'''
         args = parser.parse_args()
-        return (Post.objects.order_by(args['sort'])
+        return (Post.objects.filter(published__ne=None).order_by(args['sort'])
```
Comment (abulte) [inline udata/core/post/api.py:75]: no `visible()` helper on Post model? Maybe it's time to add one :-)

Comment (quaxsze) [reply udata/core/post/api.py]: actually there is one already :)

File: udata/core/post/factories.py [hunk 2]
```diff
@@ -7,6 +7,15 @@
 from .models import Post
 
 
+class UnpublishedPostFactory(ModelFactory):
```
Comment (abulte) [inline udata/core/post/factories.py:10]: I'm not sure you need this, can't you use PostFactory with `published=None`?

**Review by abulte — APPROVED** [review]

File: udata/core/post/factories.py (hunk 2 above)
Comment (quaxsze) [reply udata/core/post/factories.py]: Yes my bad

**Pull request merged**